Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM rocker/r-ver:4.3.1
+
+ENV RENV_CONFIG_REPOS_OVERRIDE "https://cloud.r-project.org/"
+ENV RENV_PATHS_LIBRARY renv/library
```

Comment:
I copied these variables over from [the GitLab version](https://gitlab.com/ccao-data-science---modeling/models/ccao_res_avm/-/blob/master/.gitlab-ci.yml?ref_type=heads#L22-24) of the workflow, but I'm not 100% confident they're necessary.

Reply:
`RENV_PATHS_LIBRARY` is likely necessary. If we want to setup package caching using Actions instead of (or in addition to) the Docker build cache, then we should set the other ones too. `RENV_CONFIG_REPOS_OVERRIDE` was the product of some weird bugs on GitLab's runners that I now can't recall.

**suggestion:** We _might_ want to set `RENV_CONFIG_REPOS_OVERRIDE` to point to the [Posit package mirror](https://www.r-bloggers.com/2023/07/posit-package-manager-for-linux-r-binaries/), since they have pre-built linux binaries. That would speed up our build times a lot.